language/documentation/book/src/address.md

```diff
@@ -6,17 +6,59 @@ Although an `address` is a 128 bit integer under the hood, Move addresses are in
 
 You can use runtime address values (values of type `address`) to access resources at that address. You *cannot* access modules at runtime via address values.
 
-## Literals
+## Addresses and Their Syntax
 
-`address` literals are 16-byte hex literals, i.e. `0x<hex encoded value>`. For convenience, leading `0`s are added for literals that are too short.
+Addresses come in two flavors, named or numerical. The syntax for a named address follows the
+same rules for any named identifier in Move. The syntax of a numerical address is not restricted
+to hex-encoded values, and any valid [`u128` numerical value](./integers.md) can be used as an
+address value, e.g., `42`, `0xCAFE`, and `2021` are all valid numerical address
+literals.
+
+To distinguish when an address is being used in an expression context or not, the
+syntax when using an address will differ slightly depending on the context it's
+being used in:
+* When an address is used as an expression the address must be prefixed by the `@` character, i.e., `@`[`<numeric_value>`](./integers.md)` or `@<named_address_identifier>`.
```

Comment:
👍

Reply:
This looks much better! Do we link to the module section here at all?

Reply:
We don't here, but I just updated below in the part on named addresses to point to the module section.